File: backend_parcial/models/Plan.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const PlanSchema = new Schema({
  name: { type: String, required: true },
  motive: { type: String },
  targetAmount: { type: Number, required: true },
  months: { type: Number, required: true },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Plan', PlanSchema);

File: app/src/main/java/com/example/plan_de_ahorro/data/model/Plan.kt
package com.example.plan_de_ahorro.data.model

import com.google.gson.annotations.SerializedName

data class Plan(
    @SerializedName("_id") val id: String? = null,
    val name: String,
    val motive: String,
    val targetAmount: Double,
    val months: Int,
    val createdAt: String? = null,
    val members: List<Member>? = null,
    val payments: List<Payment>? = null,
    val totalCollected: Double? = null
)
